src/TextInputWrapper.cpp

```diff
@@ -0,0 +1,94 @@
+#include <Gosu/Gosu.hpp>
+
+namespace Gosu
+{
+  class TextInputForWrapper : public Gosu::TextInput
+  {
+    public:
+      TextInputForWrapper();
+      std::string filter(std::string text) const override;
+      const char* (*filter_callback)(const char*) = nullptr;
```

Comment:
I'm not sure how Ruby or Crystal would manage the lifetime of the `const char*` that their functions must return :/ Maybe you could hack this with more indirection, e.g. TextInputForWrapper::filter calls a callback, and that callback calls  `Gosu_TextInput_complete_filter(@text_input, "filter_result")` with the result, and then control returns into `TextInputForWrapper::filter` where this stored value is used as the C++ return value. (Oof. Does that make sense?!)

Reply:
Let's see if I got this straight:
`TextInput::filter` calls `filter_callback(text)` and that callback writes to, say, `TextInput::filter_result` and `TextInput::filter` reads from there instead of directly from `filter_callback()`.

Seems simple enough.

Reply:
Right. Except that the writing to `TextInput::filter_result` will have to happen through another function/method because Crystal cannot save data in a std::string instance variable directly.

Reply:
Implemented https://github.com/gosu/gosu/pull/501/commits/f390785f66f76dd60097785852d88d5f9850ef45